src/client/dfs/dfs.c

```diff
@@ -1805,9 +1809,10 @@ dfs_remove(dfs_t *dfs, dfs_obj_t *parent, const char *name, bool force,
 	return rc;
 }
 
-int
-dfs_lookup(dfs_t *dfs, const char *path, int flags, dfs_obj_t **_obj,
-	   mode_t *mode, struct stat *stbuf)
+static int
+lookup_rel_path(dfs_t *dfs, dfs_obj_t *root, const char *path, int flags,
+		dfs_obj_t **_obj, mode_t *mode, struct stat *stbuf,
+		size_t depth)
```

Comment:
@ashleypittman This is the function I mentioned before. In order to handle symlink dereferencing, I needed to be able to lookup a path relative to some parent, which is what this function does. It's literally just `dfs_lookup`, but relative to some `root`.